### PR TITLE
Align purpose with spec using verification method and verification relationships

### PR DESCRIPTION
**Pull request by isaacJChen**

### CODE REVIEW

File: lib/core/versions/latest/DocumentComposer.ts [hunk 1]
```diff
@@ -24,10 +24,11 @@ export default class DocumentComposer {
 
     const document = didState.document as DocumentModel;
 
-    // Only populate `publicKey` if general purpose exists.
-    // Only populate `authentication` if authentication purpose exists.
-    const authentication: any[] = [];
-    const publicKeys: any[] = [];
+    // Put each public key in verificationMethod
+    // then populate the verification relationships by reference if a key has multiple purposes,
+    // or populate the verification relattionship by full description if a key has only 1 purpose
```
Comment (thehenrytsai) [inline lib/core/versions/latest/DocumentComposer.ts:29]: Sanity: I thought Daniel has confirm that we do by reference always?

File: lib/core/versions/latest/DocumentComposer.ts [hunk 2]
```diff
@@ -24,10 +24,11 @@ export default class DocumentComposer {
...
+    const verificationRelationships: Map<string, Array<object | string>> = new Map();
```
Comment (thehenrytsai) [inline lib/core/versions/latest/DocumentComposer.ts:30]:
```suggestion
    const verificationRelationships: Map<string, string[]> = new Map();
```

File: tests/core/Resolver.spec.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ describe('Resolver', () => {
       const published = true;
       const didState = await resolver.resolve(didUniqueSuffix) as DidState;
       const resultingDocument = DocumentComposer.transformToExternalDocument(didState, `did:sidetree:${didUniqueSuffix}`, published);
-      fixtureDriftHelper(resultingDocument, afterUpdate, 'resolution/afterUpdate.json', OVERWRITE_FIXTURES);
+      fixtureDriftHelper(resultingDocument, afterUpdate, 'resolution/afterUpdate.json', false);
```
Comment (thehenrytsai) [inline tests/core/Resolver.spec.ts:92]: Sanity: did you intend to make these false, if so, why?

**Review by thehenrytsai — CHANGES_REQUESTED** [review]
Minor changes requested.

File: tests/core/Resolver.spec.ts (hunk 3 above)
Comment (isaacJChen) [reply tests/core/Resolver.spec.ts:92]: Good catch. Not intended.

File: lib/core/versions/latest/DocumentComposer.ts (hunk 1 above)
Comment (isaacJChen) [reply lib/core/versions/latest/DocumentComposer.ts:29]: Old comments before we made the decision, will update.

**Review by thehenrytsai — APPROVED** [review]

**Pull request merged**